woocommerce/woocommerce
# CLI tool and GH workflow to highlight PR changes

### All Submissions:






### Changes proposed in this Pull Request:

CLI tool to analyze PRs for:

- Templates updated
- New actions and filters

GH workflow:

- Runs this CLI tool and insert a comment with all results

This tool will check for the plugin version and compare against hooks and templates changes found, to make sure all are new changes introduced in the latest release.

Fixes https://github.com/woocommerce/woocommerce/issues/29814

### How to test the changes in this Pull Request:

1. Check out this branch
2. Run `pnpm install`
3. Create a new branch and introduce some new hook with a `@since (make sure is the same as your `woocommerce.php`)
4. Make changes in some templates, and in one of the templates you changed make an update to the `@version` docblock (make sure is the same as your `woocommerce.php`).
5. Commit and push your branch to GH.
6. Run `./tools/code-analyzer/bin/dev analyzer BRANCH_TO_COMPARE` 
7. Fork this repository and merge this PR into trunk on your forked repository.
8. Edit line 18 of `.github/workflows/pr-highlight-changes.yml`, to include your forked repo `./tools/code-analyzer/bin/dev analyzer "$GITHUB_HEAD_REF" -o github `-s <github-org/repo>` (replace with your forked repo slug).
9. Go to the actions tab and enable it.
10. Go to your repo settings and set a GH token to make the commits in your forked repository using the `GITHUB_HEAD_REF` ID.
11. Create a new PR in your forked repository including some new hooks, and making template changes like in step `4`.
12. Check if the workflow is running correctly and if includes comments to the PR.

```
./tools/code-analyzer/bin/dev analyzer add/changes -s claudiosanches/woocommerce 
```

### Other information:




### Changelog entry

None

### FOR PR REVIEWER ONLY:
